opal/threads/wait_sync.c

```diff
@@ -89,20 +91,23 @@ int ompi_sync_wait_mt(ompi_wait_sync_t *sync)
     }
     OPAL_THREAD_ADD_FETCH32(&num_thread_in_progress, -1);
 
-    assert(sync == wait_sync_list);
+    if (NULL != wait_sync_list) {
+        assert(sync == wait_sync_list);
```

Comment:
My code crashed with this assert. I think we have to rethink this PR. Here is the problem..
1. This assert is problematic because maybe you wake up more than 1 thread (more than 1 sync completed)
2. The check of `opal_max_thread_in_progress` to decide if we are going to update the list is open to race condition. (read after write on active thread count, etc)